### PR TITLE
Yoom/tree render class

### PR DESCRIPTION
**Pull request by yoomlam**
This PR is an improved version of PR #13063.
* moves most of the new code to a class so as not to pollute Task and Appeal classes which include the module.
* better usability for configuring output options. 

### Description
An easier way to inspect task trees with fewer keystrokes.

Motivations:
* easier column-aligned printout for comparing attributes across tasks,
* attribute dereference chaining and flexible transforms,
* customizable column headings,
* and more visually appealing.

See usage instructions at https://github.com/department-of-veterans-affairs/caseflow/wiki/Task-Tree-Render

### Acceptance Criteria
- [ ] All instructions work as described

### Testing Plan
1. Run through instructions and test custom render settings

### CODE REVIEW

File: app/models/concerns/task_tree_metadata.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  def calculate_max_name_length(task, depth = 0)
+    task_label_length = (INDENT_SIZE * depth) + task.class.name.length
+    task.children.map do |child|
+      calculate_max_name_length(child, depth + 1)
+    end.append(task_label_length).max
+  end
+
+  def appeal_children(appeal)
+    roottask_ids = appeal.tasks.where(parent_id: nil).pluck(:id)
+    # Can the following be expressed using `where` so that it returns an AssociationRelation of Tasks?
```
Comment (lomky) [inline app/models/concerns/task_tree_metadata.rb]: After a few minutes trying out some ways, I don't _think_ so.

File: app/models/concerns/task_tree_metadata.rb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        configure_headings_using_labels(col_labels, col_metadata, col_keys)
+      else
+        configure_headings_using_transform(col_metadata)
+      end
+
+      update_col_widths_to_fit_col_labels(col_metadata)
+    end
+  end
+
+  # Calculate column widths using rows only (not column heading labels)
```
Comment (lomky) [inline app/models/concerns/task_tree_metadata.rb]: I really appreciate these comments. Any reason not to add them to all the functions on the other files too?

File: spec/models/concerns/task_tree_render_module_spec.rb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      kwargs.delete(:renderer) && fail("Use other approach to allow 'renderer' named parameter!")
+      renderer = TaskTreeRenderModule.new_renderer.tap do |r|
+        r.compact
+        r.config.default_atts = [:id, :status, :ASGN_TO, :UPD_DATE]
+      end
+      renderer.tree_str(obj, *atts, **kwargs)
+    end
+
+    it "prints all tasks" do
+      # puts tree2 @appeal, :id, :status
```
Comment (lomky) [inline spec/models/concerns/task_tree_render_module_spec.rb]: dead code to cleanup

File: spec/models/concerns/task_tree_render_module_spec.rb [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    it "highlights self task with an asterisk" do
+      task_to_highlight = @appeal.tasks.sample
+      _rows_hash, metadata = task_to_highlight.tree_hash(" ", :id, :status)
+      check_for_highlight(metadata, task_to_highlight)
+    end
+
+    it "highlights specified task with an asterisk, even if no columns are specified" do
+      task_to_highlight = @appeal.tasks.sample
+      # @appeal.treee(:id, :status, highlight: task_to_highlight.id)
+      # task_to_highlight.treee(:id, :status, highlight: task_to_highlight.id)
```
Comment (lomky) [inline spec/models/concerns/task_tree_render_module_spec.rb]: deadcode to cleanup

File: spec/models/concerns/task_tree_render_module_spec.rb [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    @appeal = create(:appeal, :with_post_intake_tasks)
+    root = @appeal.root_task
+    create(:ama_judge_task, appeal: @appeal, parent: root, created_at: 1.day.ago)
+    create(:ama_attorney_task, parent: root, appeal: @appeal)
+    create(:ama_attorney_task, appeal: @appeal)
+  end
+
+  context ".tree is called on an appeal" do
+    it "returns all tasks for the appeal" do
+      # puts @appeal.tree
```
Comment (lomky) [inline spec/models/concerns/task_tree_render_module_spec.rb]: deadcode to cleanup

File: app/models/concerns/task_tree_renderer.rb [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+  class << self
+    def send_chain(initial_obj, methods)
+      methods.inject(initial_obj) do |obj, method|
+        obj.respond_to?(method) ? obj.send(method) : nil
+      end
+    end
+  end
+
+  def ansi
```
[the remark this answers is not in the document]
Comment (yoomlam) [reply app/models/concerns/task_tree_renderer.rb]: I was trying to keep these method names short to reduce keystrokes. Hmmm.

File: app/models/concerns/task_tree_metadata.rb (hunk 2 above)
Comment (yoomlam) [reply app/models/concerns/task_tree_metadata.rb]: I'll add comments to the non-obvious ones. Reason I didn't: doesn't seem like the code base has any code comments so I didn't want to unnecessarily pollute it.

**Comment by yoomlam** [comment]
Extracted generic code into a gem: https://github.com/department-of-veterans-affairs/console-tree-renderer

File: Gemfile [hunk 7]
```diff
@@ -14,6 +14,7 @@ gem "bootsnap", require: false
 gem "business_time", "~> 0.9.3"
 gem "caseflow", git: "https://github.com/department-of-veterans-affairs/caseflow-commons", ref: "ffb77dd0395cbd5b7c1a5729f7f8275b5ec681fa"
 gem "connect_vbms", git: "https://github.com/department-of-veterans-affairs/connect_vbms.git", ref: "a97a56f72d4a08f634c7f569fdcc83615362a8b2"
+gem "console_tree_renderer", git: "https://github.com/department-of-veterans-affairs/console-tree-renderer.git", tag: "v0.1.0"
```
Comment (yoomlam) [inline Gemfile]: Use new repo https://github.com/department-of-veterans-affairs/console-tree-renderer as a gem.

File: .reek.yml [hunk 8]
```diff
@@ -36,6 +36,7 @@ detectors:
     exclude:
       - HearingSerializerBase
       - LegacyDocket
+      - TaskTreeRenderModule
```
Comment (yoomlam) [inline .reek.yml]: Reek doesn't like my uses of `*atts, **kwargs` like in `def tree(*atts, **kwargs)`

**Pull request merged**